src/analysis/retail/shaman/restoration/modules/talents/totemic/SurgingTotem.tsx

```diff
@@ -0,0 +1,391 @@
+import { defineMessage, Trans } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import TALENTS, { TALENTS_SHAMAN } from 'common/TALENTS/shaman';
+import { SpellIcon } from 'interface';
+import { SpellLink } from 'interface';
+import { TooltipElement } from 'interface';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { HealEvent, CastEvent, RemoveBuffEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import Combatants from 'parser/shared/modules/Combatants';
+import CastEfficiencyBar from 'parser/ui/CastEfficiencyBar';
+import { GapHighlight } from 'parser/ui/CooldownBar';
+import StatisticBox, { STATISTIC_ORDER } from 'parser/ui/StatisticBox';
+import { GUIDE_CORE_EXPLANATION_PERCENT } from '../../../Guide';
+import { BoxRowEntry } from 'interface/guide/components/PerformanceBoxRow';
+import { didMoteExpire } from '../../../normalizers/CastLinkNormalizer';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import CastSummaryAndBreakdown from 'interface/guide/components/CastSummaryAndBreakdown';
+import { HEALING_RAIN_TARGETS } from '../../../constants';
+
+// 50 was too low, 100 was too high
+// had no issues with 85ms
+const BUFFER_MS = 85;
```

Comment:
as a note these things tend to be latency-sensitive. so unfortunately there will never be a perfect number since some people will have better or worse ping :(

Reply:
I suppose so, I took these numbers from the Healing rain module, and played around with various other values but this looks like a good middle ground